graylog2-server/src/main/java/org/graylog/security/shares/EntitySharesService.java

```diff
@@ -169,13 +167,21 @@ public EntityShareResponse updateEntityShares(GRNType grnType, String id, Entity
 
     /**
      * Share / unshare an entity with one or more grantees.
-     * The grants in the request are created or, if they already exist, updated.
+     * The grants in the request are created or, if they already exist, updated. Any dependent entities - as
+     * provided by the {@link DependentEntitiesResolver} - are also updated.
      *
      * @param ownedEntity the target entity for the updated grants
      * @param request     the request containing grantees and their capabilities
      * @param sharingUser the user executing the request
      */
     public EntityShareResponse updateEntityShares(GRN ownedEntity, EntityShareRequest request, User sharingUser) {
+        final EntityShareResponse result = updateEntitySharesWithoutDependencies(ownedEntity, request, sharingUser);
+        return result.toBuilder().syncedDependentEntities(
+                        resolveImplicitGrants(ownedEntity, request, sharingUser))
+                .build();
+    }
+
+    private EntityShareResponse updateEntitySharesWithoutDependencies(GRN ownedEntity, EntityShareRequest request, User sharingUser) {
```

Comment:
We rename the interface, so we should also rename this method.

```suggestion
    private EntityShareResponse updateEntitySharesWithoutSyncedEntities(GRN ownedEntity, EntityShareRequest request, User sharingUser) {
```

Reply:
I liked Kai's suggestion: `updatePrimaryEntityShares`

Reply:
It's Kay, but I am fine with that as well.